internal/openshift/controller_test.go

```diff
@@ -52,6 +52,10 @@ func Test_handle_build(t *testing.T) {
 
 	err := controller.HandleBuild(obj)
 	assert.NoError(t, err)
+
+	obj.Type = "DELETED"
```

Comment:
can you add a test for  the case where the type isn't `"DELETED"` ?

Reply:
There is already one test above that is not DELETED, but let me add tests for all WatchEvents https://github.com/kubernetes/kubernetes/blob/932e657d5d0e98624fb9907ea34a55ac604253d3/pkg/watch/json/types.go#L35

Reply:
Added extra WatchEvent types. but whatever tests that we have don't seem to do much. Do you want me to add some better tests?
(sorry for rebase. I made a mistake while using git. Had to do it.)

Reply:
>  Do you want me to add some better tests? 

Yes, please. the current tests do not assert changes to controllers internals i.e. other than testing that it does not result in an `error`. Ideally you would want tests to invoke the method and assert that the internal state changes are as expected.